### ISSUE TITLE
Kubernetes / Helm package for Perplexica

### ISSUE THREAD

**Issue by michael-willingham** [ItzCrazyKns/Perplexica#145]
It would be great to see a Helm chart for this to run easily on Kubernetes! A bit busy with work at the moment, but might be able to take a look at it in the coming weeks.

**Comment by michael-willingham** [comment]
Sadly I have not had the time! Work and life have kept me quite busy. This might be better off if someone else could take a look tbh

**Comment by delta-whiplash** [comment]
ok no worries could you just automate at least the docker container creation with the github actions template.
With this (5min of clicks) you can automate the build and publish of your docker image on this github repo soo I can write a kubernetes deployment to do some testings 

you can do it like this : 
https://github.com/ItzCrazyKns/Perplexica/new/main?filename=.github%2Fworkflows%2Fdocker-publish.yml&workflow_template=ci%2Fdocker-publish

**Comment by michael-willingham** [comment]
This is not my repo, I'm just a user! That's a question for @ItzCrazyKns

**Comment by timoa** [comment]
Hey @ItzCrazyKns!
I just started to work on the Helm Chart based on the current Docker Compose file.
I'm closed to push the PR. Just need to finish the PV&PVC.

**Comment by timoa** [comment]
Hi, @ItzCrazyKns! I was able to deploy Perplexica a few minutes ago on my Kubernetes cluster 🎉

<img width="569" alt="Screenshot 2024-10-24 at 07 56 35" src="https://github.com/user-attachments/assets/2393615b-200a-47bd-9d3c-3d29bdedabbe">

But the CORS config is restricting access to the backend API to http://127.0.0.1, where I deployed it with a public URL:

<img width="482" alt="Screenshot 2024-10-24 at 07 48 50" src="https://github.com/user-attachments/assets/9a84846b-b0ff-4ffe-8c7f-e7762874af99">

To solve that, I will check how the CORS is configured on the backend side.

**Comment by delta-whiplash** [comment]
Nice work

**Comment by Eleven-am** [comment]
here is my deployments 

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: searxng-deployment
  namespace: search
  labels:
    app: searxng-deployment
spec:
  replicas: 1
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1
  selector:
    matchLabels:
      app: searxng-deployment
  template:
    metadata:
      name: searxng-deployment
      labels:
        app: searxng-deployment
    spec:
      restartPolicy: Always
      containers:
        - name: searxng
          image: searxng/searxng:latest
          imagePullPolicy: Always
          ports:
            - containerPort: 8080
              name: searxng
              protocol: TCP
          env:
            - name: PUID
              value: "1005"
            - name: PGID
              value: "1005"
            - name: TZ
              value: "Europe/Paris"
          volumeMounts:
            - name: searxng-config
              mountPath: /etc/searxng
      volumes:
        - name: searxng-config
          persistentVolumeClaim:
            claimName: searxng-data
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: backend-deployment
  namespace: search
  labels:
    app: backend-deployment
spec:
  replicas: 1
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1
  selector:
    matchLabels:
      app: backend-deployment
  template:
    metadata:
      name: backend-deployment
      labels:
        app: backend-deployment
    spec:
      restartPolicy: Always
      containers:
        - name: backend
          image: itzcrazykns1337/perplexica-backend:main
          imagePullPolicy: Always
          ports:
            - containerPort: 3001
              name: backend
              protocol: TCP
          env:
            - name: PUID
              value: "1005"
            - name: PGID
              value: "1005"
            - name: TZ
              value: "Europe/Paris"
            - name: SEARXNG_API_URL
              value: searxng-service.search.svc.cluster.local
          volumeMounts:
            - name: backend-dbstore
              mountPath: /home/perplexica/data
            - name: uploads
              mountPath: /home/perplexica/uploads
            - name: config
              mountPath: /home/perplexica/config.toml
              subPath: config.toml
      volumes:
        - name: backend-dbstore
          persistentVolumeClaim:
            claimName: backend-data
        - name: uploads
          persistentVolumeClaim:
            claimName: backend-media
        - name: config
          configMap:
            name: perplexica-config
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: frontend-deployment
  namespace: search
  labels:
    app: frontend-deployment
spec:
  replicas: 1
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1
  selector:
    matchLabels:
      app: frontend-deployment
  template:
    metadata:
      name: frontend-deployment
      labels:
        app: frontend-deployment
    spec:
      restartPolicy: Always
      containers:
        - name: frontend
          image: itzcrazykns1337/perplexica-frontend:main
          imagePullPolicy: Always
          ports:
            - containerPort: 3001
              name: frontend
              protocol: TCP
          env:
            - name: NEXT_PUBLIC_API_URL
              value: "https://perplexica-backend.domain/api"
            - name: NEXT_PUBLIC_WS_URL
              value: "wss://perplexica-backend.domain"
```

for some reason it is still trying to hit 127.0.0.1:3001, is there a way to fix this issue

**Comment by mitchross** [comment]
@timoa  its because the docker build file hard codes the args

```
  perplexica-frontend:
    build:
      context: .
      dockerfile: app.dockerfile
      args:
        - NEXT_PUBLIC_API_URL=http://127.0.0.1:3001/api
        - NEXT_PUBLIC_WS_URL=ws://127.0.0.1:3001
```
        
        
        ill see if i can make a fix so its more k8s friendly

**Comment by mitchross** [comment]
https://github.com/ItzCrazyKns/Perplexica/pull/546

**Comment by chrisjaimon2012** [comment]
@timoa wondering if you got a chance to test it after #546

**Comment by flefevre** [comment]
Does anyone succeed in deploying perplixica with helmchart ?

**Comment by mitchross** [comment]
@flefevre  you dont need helm. I have it deployed and working.  https://github.com/mitchross/talos-argocd-proxmox/tree/main/my-apps/ai/perplexica

**Comment by flefevre** [comment]
Thanks @mitchross but in our research institute we deploy applications only with kubernetes for load balancing, scaleup, secret management or volume with S3 etc.
That's why I am looking for an helmchart. It is mandatory for applications that become a daily service at the heart of research.
Hope the team of perplixica could look at since they've already done a so beautiful and useful application.

**Comment by mitchross** [comment]
> Thanks [@mitchross](https://github.com/mitchross) but in our research institute we deploy applications only with kubernetes for load balancing, scaleup, secret management or volume with S3 etc. That's why I am looking for an helmchart. It is mandatory for applications that become a daily service at the heart of research. Hope the team of perplixica could look at since they've already done a so beautiful and useful application.

What I linked is kubernetes...

**Comment by wwagops** [comment]
I needed to deploy Vane with Helm on a Kubernetes cluster, so I went ahead and wrote a chart for it and opened a PR: #1158

It's a single chart that covers both official images through a `variant` switch:

- `full` (default) → `itzcrazykns1337/vane:latest`, with SearXNG bundled in the image
- `slim` → `itzcrazykns1337/vane:slim-latest`, with SearXNG either deployed in-cluster by the chart (`searxng.deploy=true`) or pointed at an existing instance (`searxng.url`)

It ships persistence for the config/SQLite database and uploads, plus Ingress, HPA, probes, and a `values.schema.json` for install-time validation. It works out of the box:

```bash
helm install vane ./charts/vane
```

Feedback and reviews welcome — hopefully this helps everyone here who has been waiting on a Helm chart. 🙂
